pkg/cmd/accept/exec.go

```diff
@@ -0,0 +1,161 @@
+	alreadyProvidedCluster := make(map[string]bool)
+	clusters := make([]string, 0)
+	if o.clusters != "" {
+		cs := strings.Split(o.clusters, ",")
+		for _, c := range cs {
+			if _, ok := alreadyProvidedCluster[c]; !ok {
+				alreadyProvidedCluster[c] = true
+				clusters = append(clusters, strings.TrimSpace(c))
+			}
+		}
+		o.values.clusters = clusters
+	} else {
+		return fmt.Errorf("values or name are missing")
+	}
+
+	return nil
+}
+
+func (o *Options) validate() error {
+	return nil
+}
+
+func (o *Options) run() error {
+	client, err := helpers.GetControllerRuntimeClientFromFlags(o.ConfigFlags)
+	if err != nil {
+		return err
+	}
+	return o.runWithClient(client)
+}
+
+func (o *Options) runWithClient(client crclient.Client) error {
+	for _, clusterName := range o.values.clusters {
+
+		csrs := &certificatesv1.CertificateSigningRequestList{}
+		ls := labels.SelectorFromSet(labels.Set{
+			clusterLabel: clusterName,
+		})
+		err := client.List(context.TODO(),
+			csrs,
+			&crclient.ListOptions{
+				LabelSelector: ls,
+			})
+		if err != nil {
+			return err
+		}
+		var csr *certificatesv1.CertificateSigningRequest
+		for _, item := range csrs.Items {
+			//Does not have the correct name prefix
+			if !strings.HasPrefix(item.Spec.Username, userNameSignaturePrefix) {
+				continue
+			}
+			//Check groups
+			var group string
+			for _, g := range item.Spec.Groups {
+				if g == groupName {
+					group = g
+					break
+				}
+			}
+			//Does not contain the correct group
+			if len(group) == 0 {
+				continue
+			}
+			//Check if already approved or denied
+			done := false
+			for _, c := range item.Status.Conditions {
+				if c.Type == certificatesv1.CertificateApproved || c.Type == certificatesv1.CertificateDenied {
+					done = true
+					break
+				}
+			}
+			if done {
+				continue
+			}
+			csr = &item
+			break
+		}
+
+		if csr != nil {
+			if csr.Status.Conditions == nil {
+				csr.Status.Conditions = make([]certificatesv1.CertificateSigningRequestCondition, 0)
+			}
+
+			csr.Status.Conditions = append(csr.Status.Conditions, certificatesv1.CertificateSigningRequestCondition{
+				Status:         corev1.ConditionTrue,
+				Type:           certificatesv1.CertificateApproved,
+				Reason:         fmt.Sprintf("%sApprove", helpers.GetExampleHeader()),
+				Message:        fmt.Sprintf("This CSR was approved by %s certificate approve.", helpers.GetExampleHeader()),
+				LastUpdateTime: metav1.Now(),
+			})
+
+			kubeClient, err := o.factory.KubernetesClientSet()
+			if err != nil {
+				return err
+			}
+			signingRequest := kubeClient.CertificatesV1().CertificateSigningRequests()
+			if _, err := signingRequest.UpdateApproval(context.TODO(), csr.Name, csr, metav1.UpdateOptions{}); err != nil {
+				return err
+			}
+
+			fmt.Printf("CSR %s approved\n", csr.Name)
+		} else {
+			fmt.Printf("no CSR to approve for cluster %s\n", clusterName)
+		}
+
+		mc := &unstructured.Unstructured{}
```

Comment:
I would prefer to use type here instead of unstructured. If you are using client from controller-runtime. I think you can add the  scheme in it and use the cluster type. I would want this as a followup.